hbase/src/main/java/org/apache/zeppelin/hbase/HbaseInterpreter.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.hbase;
+
+import org.apache.commons.exec.*;
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterPropertyBuilder;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.jruby.embed.LocalContextScope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.jruby.embed.ScriptingContainer;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Created by rajatv on 4/22/15.
+ */
+public class HbaseInterpreter extends Interpreter {
+  Logger logger = LoggerFactory.getLogger(HbaseInterpreter.class);
+  int commandTimeOut = 600000;
+  ScriptingContainer scriptingContainer;
+
+  StringWriter writer;
+
+  static {
+    Interpreter.register("hbase", "hbase", HbaseInterpreter.class.getName(),
+        new InterpreterPropertyBuilder()
+            .add("hbase.home", "/usr/lib/hbase/", "Installation dir. of Hbase")
+            .add("hbase.ruby.sources", "lib/ruby",
+                "Path to Ruby scripts relative to 'hbase.home'")
+            .add("hbase.irb.load", "true", "Load hirb. Optional for testing only")
+            .build());
+  }
+
+  public HbaseInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {
+    String hbase_home = getProperty("hbase.home");
+    String ruby_src = getProperty("hbase.ruby.sources");
+    String abs_ruby_src = hbase_home + ruby_src;
+
+    logger.info("Home:" + hbase_home);
+    logger.info("Ruby Src:" + ruby_src);
+
+    Properties props = System.getProperties();
+    props.setProperty("hbase.ruby.sources", abs_ruby_src);
+    this.scriptingContainer  = new ScriptingContainer(LocalContextScope.SINGLETON);
+    List<String> paths = new ArrayList<>(Arrays.asList(abs_ruby_src));
+    this.writer = new StringWriter();
+    scriptingContainer.setOutput(this.writer);
+    this.scriptingContainer.setLoadPaths(paths);
+    scriptingContainer.setCompatVersion(org.jruby.CompatVersion.RUBY1_9);
+    if (Boolean.parseBoolean(getProperty("hbase.irb.load"))) {
+      try {
+        InputStream in = getClass().getResourceAsStream("/hbase/bin/hirb.rb");
+        scriptingContainer.runScriptlet(in, "/hbase/bin/hirb.rb");
+        in.close();
+      } catch (NullPointerException | IOException e) {
+        logger.error("Open failed:", e);
+      }
+    }
+  }
+
+  @Override
+  public void close() {}
```

Comment:
You may want to release the resources here with `#terminate()`: https://github.com/jruby/jruby/blob/2187665d26d227d8f0b0882c762c7b7485efe642/core/src/main/java/org/jruby/embed/ScriptingContainer.java#L1853